src/coreclr/vm/assembly.cpp

```diff
@@ -1936,9 +1937,426 @@ Assembly::EnumMemoryRegions(CLRDataEnumMemoryFlags flags)
+    // running before Fixup. A concrete example VS recently hit, calling GetClassLayout using
+    // a MethodTable which doesn't need restore but its parent pointer isn't fixed up yet.
+    // We've already set the rules so that profilers can't modify the member list of types in NGEN images
+    // so it doesn't matter if types are pre-loaded. We only need the guarantee that code for the
+    // loaded types won't execute yet. For NGEN images we deliver the load notification in
+    // FILE_LOAD_DELIVER_EVENTS.
+    if (!IsProfilerNotified())
+    {
+        SetProfilerNotified();
+        GetModule()->NotifyProfilerLoadFinished(S_OK);
+    }
+
+#endif
+}
+
+void Assembly::EagerFixups()
+{
+    WRAPPER_NO_CONTRACT;
+
+#ifdef FEATURE_READYTORUN
+    if (GetModule()->IsReadyToRun())
+    {
+        GetModule()->RunEagerFixups();
+
+    }
+#endif // FEATURE_READYTORUN
+}
+
+void Assembly::VtableFixups()
+{
+    WRAPPER_NO_CONTRACT;
+
+    GetModule()->FixupVTables();
+}
+
+void Assembly::FinishLoad()
+{
+    CONTRACTL
+    {
+        INSTANCE_CHECK;
+        STANDARD_VM_CHECK;
+    }
+    CONTRACTL_END;
+
+    // Must set this a bit prematurely for the DAC stuff to work
+    m_level = FILE_LOADED;
+
+    // Now the DAC can find this module by enumerating assemblies in a domain.
+    DACNotify::DoModuleLoadNotification(m_pModule);
+}
+
+void Assembly::Activate()
+{
+    CONTRACT_VOID
+    {
+        INSTANCE_CHECK;
+        PRECONDITION(IsLoaded());
+        STANDARD_VM_CHECK;
+    }
+    CONTRACT_END;
+
+    // We cannot execute any code in this assembly until we know what exception plan it is on.
+    // At the point of an exception's stack-crawl it is too late because we cannot tolerate a GC.
+    // See PossiblyUnwrapThrowable and its callers.
+    GetModule()->IsRuntimeWrapExceptions();
+
+    //
+    // Now call the module constructor.  Note that this might cause reentrancy;
+    // this is fine and will be handled by the class cctor mechanism.
+    //
+
+    MethodTable *pMT = m_pModule->GetGlobalMethodTable();
+    if (pMT != NULL)
+    {
+        pMT->CheckRestore();
+#ifdef _DEBUG
+        m_bDisableActivationCheck=TRUE;
+#endif
+        pMT->CheckRunClassInitThrowing();
+    }
+#ifdef _DEBUG
+    if (g_pConfig->ExpandModulesOnLoad())
+    {
+        m_pModule->ExpandAll();
+    }
+#endif //_DEBUG
+
+#ifdef FEATURE_READYTORUN
+    if (m_pModule->IsReadyToRun())
+    {
+        m_pModule->GetReadyToRunInfo()->RegisterUnrelatedR2RModule();
+    }
+#endif
+
+    RETURN;
+}
+
+void Assembly::Begin()
+{
+    STANDARD_VM_CONTRACT;
+
+    {
+        AppDomain::LoadLockHolder lock(AppDomain::GetCurrentDomain());
+        AppDomain::GetCurrentDomain()->AddAssembly(GetDomainAssembly());
+    }
+    // Make it possible to find this DomainAssembly object from associated BINDER_SPACE::Assembly.
+    RegisterWithHostAssembly();
+}
+
+void Assembly::RegisterWithHostAssembly()
```

Comment:
Since some of these are only used within assembly.cpp, they can be marked static and removed from the header file (in a later PR).